flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/SourceOperator.java

```diff
@@ -281,6 +290,21 @@ public void open() throws Exception {
                             watermarkStrategy, sourceMetricGroup);
         }
 
+        latencyMarerEmitter =
+                new LatencyMarkerEmitter<>(
+                        getProcessingTimeService(),
+                        getExecutionConfig().isLatencyTrackingConfigured()
+                                ? getExecutionConfig().getLatencyTrackingInterval()
+                                : getContainingTask() == null
+                                        ? 0
+                                        : getContainingTask()
+                                                .getEnvironment()
+                                                .getTaskManagerInfo()
+                                                .getConfiguration()
+                                                .getLong(MetricOptions.LATENCY_INTERVAL),
+                        config == null ? null : getOperatorID(),
```

Comment:
Is it possible that the container task and config are null, do we really need the judgement?

Reply:
IMO, it's unnecessary to check whether the task and config is null or not. But in `SourceOperatorEventTimeTest`, the `setup` method doesn't be invoked in `createTestOperator`, which causes that the task and config is null.
@piotrp , what do you think about?

Reply:
I think `SourceOperatorEventTimeTest#createTestOperator` is simply incorrect. Please check my other comments, as this might be related to using test harnesses.

Reply:
@pnowojski , the null check of the task and config is to solve the incorrect for `SourceOperatorEventTimeTest#createTestOperator`. Therefore, I would like to update this method to correct the creation of `SourceOperator`.